File: fixtures/py-linear-deps/src/service.py
"""中间层：依赖 utils，提供业务服务。"""
from .utils import clamp, fetch_data, Range


class NumberService:
    def __init__(self, value_range: Range):
        self.value_range = value_range

    def normalize(self, value: int) -> int:
        return clamp(value, self.value_range.low, self.value_range.high)

    async def load(self, url: str) -> list:
        data = await fetch_data(url)
        return [self.normalize(v) for v in data]

File: .tmp-lint-scan/proj/src/service.ts
import { clamp, Range, fetchData } from './utils';

export class NumberService {
  private range: Range;

  constructor(range: Range) {
    this.range = range;
  }

  normalize(value: number): number {
    return clamp(value, this.range.min, this.range.max);
  }

  async load(url: string): Promise<number[]> {
    const data = await fetchData<number[]>(url);
    return data.map(v => this.normalize(v));
  }
}
